### ISSUE TITLE
Fails to build on CentOS Stream 8

### ISSUE THREAD

**Issue by dupondje** [oVirt/vdsm-jsonrpc-java#32]
Current GIT master fails to build on a clean CentOS Stream 8:
```
Exception in thread "main" com.google.common.util.concurrent.ExecutionError: com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: javax/annotation/PostConstruct
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2212)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4053)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4057)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4986)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4992)
	at com.google.inject.internal.FailableCache.get(FailableCache.java:54)
	at com.google.inject.internal.ConstructorInjectorStore.get(ConstructorInjectorStore.java:49)
	at com.google.inject.internal.ConstructorBindingImpl.initialize(ConstructorBindingImpl.java:155)
	at com.google.inject.internal.InjectorImpl.initializeBinding(InjectorImpl.java:592)
	at com.google.inject.internal.AbstractBindingProcessor$Processor.initializeBinding(AbstractBindingProcessor.java:173)
	at com.google.inject.internal.AbstractBindingProcessor$Processor.lambda$scheduleInitialization$0(AbstractBindingProcessor.java:160)
	at com.google.inject.internal.ProcessedBindingData.initializeBindings(ProcessedBindingData.java:49)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:124)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:108)
	at com.google.inject.Guice.createInjector(Guice.java:87)
	at com.google.inject.Guice.createInjector(Guice.java:69)
	at com.google.inject.Guice.createInjector(Guice.java:59)
	at org.codehaus.plexus.DefaultPlexusContainer.addPlexusInjector(DefaultPlexusContainer.java:481)
	at org.codehaus.plexus.DefaultPlexusContainer.<init>(DefaultPlexusContainer.java:206)
	at org.apache.maven.cli.MavenCli.container(MavenCli.java:651)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

Seems like the COPR builds use an older/outdated CentOS version?

**Comment by dupondje** [comment]
This only seems to happen when using module hotfixes from Appstream btw.

**Comment by mwperina** [comment]
As COPR builds for [vdsm-jsonrpc-java](https://copr.fedorainfracloud.org/coprs/ovirt/ovirt-master-snapshot/package/vdsm-jsonrpc-java/) work fine, the latest version is available in [oVirt master snaphot](https://copr.fedorainfracloud.org/coprs/ovirt/ovirt-master-snapshot/).

Also when looking at the latest build job logs for vdsm-jsonrpc-java on [github](https://github.com/oVirt/vdsm-jsonrpc-java/actions/runs/5036961137/jobs/9033479377) and on [COPR](https://download.copr.fedorainfracloud.org/results/ovirt/ovirt-master-snapshot/centos-stream-8-x86_64/05939288-vdsm-jsonrpc-java/builder-live.log.gz) , I can see that latest CS8 packages are used, for example `gcc-8.5.0-19.el8` is available on CS8 while `gcc-8.5.0-18.el8` is available on Alma 8.8 (both testing installations were fully updated today).

So could you please check that on your CS8 installation there are enable the same repositories and DNF modules as we are using for oVirt builds in the [oVirt build container](https://github.com/oVirt/buildcontainer/blob/main/Dockerfile.stream8)?

**Comment by dupondje** [comment]
Seems like it was caused because my mock build env had module_hotfixes=1 instead of just using specific modules.
Builds fine now.
